exec/java-exec/src/test/java/org/apache/drill/exec/store/parquet/TestParquetReaderDecision.java

```diff
@@ -0,0 +1,99 @@
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.parquet.hadoop.metadata.ParquetMetadata;
+import org.apache.parquet.hadoop.ParquetFileReader;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.ArrayList;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+
+/**
+ * DRILL-5797 introduces more granularity on new reader use cases. This test is aimed at
+ * checking correctness of function used for new reader usage decision making.
+ */
+public class TestParquetReaderDecision {
+
+  private static final String path = "src/test/resources/store/parquet/complex/complex.parquet";
+  private static Configuration conf;
+  private static ParquetMetadata footer;
+
+  @BeforeClass
+  public static void setUpBeforeClass() throws Exception {
+    conf = new Configuration();
+
+    try {
+      footer = ParquetFileReader.readFooter(conf, new Path(path));
+    } catch (IOException ioe) {
+      fail("Could not read Parquet file '" + path + "', error message: " + ioe.getMessage()
+          + " cwd: " + Paths.get(".").toAbsolutePath().normalize().toString());
+      throw(ioe);
+    }
+  }
+
+  @Test
+  public void testParquetReaderDecision() {
+    List<SchemaPath> caseOldReader1 = new ArrayList<>();
+    List<SchemaPath> caseOldReader2 = new ArrayList<>();
+    List<SchemaPath> caseOldReader3 = new ArrayList<>();
+    List<SchemaPath> caseNewReader1 = new ArrayList<>();
+    List<SchemaPath> caseNewReader2 = new ArrayList<>();
+    List<SchemaPath> caseNewReader3 = new ArrayList<>();
+
+    SchemaPath topNestedPath = SchemaPath.getCompoundPath("marketing_info");
+    SchemaPath nestedColumnPath = SchemaPath.getCompoundPath("marketing_info", "camp_id");
+    SchemaPath topPath1 = SchemaPath.getCompoundPath("trans_id");
+    SchemaPath topPath2 = SchemaPath.getCompoundPath("amount");
+    SchemaPath nonExistentPath = SchemaPath.getCompoundPath("nonexistent");
+
+    caseOldReader1.add(topNestedPath);
+    caseOldReader2.add(nestedColumnPath);
+    caseOldReader3.add(topPath1);
+    caseOldReader3.add(nestedColumnPath);
+
+    caseNewReader1.add(topPath1);
+    caseNewReader2.add(topPath1);
+    caseNewReader2.add(topPath2);
+
+    assertTrue("Complex column not detected",
+        AbstractParquetScanBatchCreator.containsComplexColumn(footer, caseOldReader1));
+    assertTrue("Complex column not detected",
+        AbstractParquetScanBatchCreator.containsComplexColumn(footer, caseOldReader2));
+    assertTrue("Complex column not detected",
+        AbstractParquetScanBatchCreator.containsComplexColumn(footer, caseOldReader3));
+
+    assertFalse("No complex column, but complex column is detected",
+        AbstractParquetScanBatchCreator.containsComplexColumn(footer, caseNewReader1));
```

Comment:
Though you have made method public static for testing, it's good practice, move method to utilities then. Semantically `AbstractParquetScanBatchCreator` definitely does not need to have this method as public static. As I have mentioned before `isComplex` and `containsComplexColumn` should be one method, might be utility one to ease testing.

Reply:
`containsComplexColumn` method moved to ParquetReaderUtility class.